spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2734,4 +2734,28 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("test integral divide overflow for decimal") {
+    // decimal support requires Spark 3.4 or later
+    assume(isSpark34Plus)
+    if (isSpark40Plus) {
+      Seq(true, false)
+    } else
+      {
+        // ansi mode only supported in Spark 4.0+
+        Seq(false)
+      }.foreach { ansiMode =>
+        withSQLConf(SQLConf.ANSI_ENABLED.key -> ansiMode.toString) {
```

Comment:
ANSI support in Comet is currently experimental and has to be enabled by also setting the internal config `spark.comet.ansi.enabled=true`.

Reply:
In spark 4, `spark.comet.ansi.enabled` defaults to `true`:

https://github.com/apache/datafusion-comet/blob/382ac938f2d10666eefc08ec5c1c82025ddf3726/common/src/main/scala/org/apache/comet/CometConf.scala#L578-L585

https://github.com/apache/datafusion-comet/blob/382ac938f2d10666eefc08ec5c1c82025ddf3726/common/src/main/spark-4.0/org/apache/comet/shims/ShimCometConf.scala#L24